batch/batch/driver/main.py

```diff
@@ -372,6 +372,35 @@ async def job_started(request, instance):
     return await asyncio.shield(job_started_1(request, instance))
 
 
+async def billing_update_1(request, instance):
+    db: Database = request.app['db']
+
+    body = await request.json()
+    args = []
+    for job in body:
+        args.append((job['batch_id'], job['job_id'], job['attempt_id'], job['start_time'], job['rollup_time']))
+
+    await db.execute_many(
+        '''
+INSERT INTO attempts (batch_id, job_id, attempt_id, start_time, rollup_time)
+VALUES (%s, %s, %s, %s, %s) AS new
+ON DUPLICATE KEY UPDATE
+  start_time = new.start_time, rollup_time = new.rollup_time;
```

Comment:
I agree with you that it would be nice to not have to do an insert here, but this is the easiest way to accomplish updating multiple rows with varying values in MySQL.